### PR TITLE
Pure package & base tests

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Add test for the package without extras

**TODO**

- [x] add doctest to run Trainer with single epoch

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
great idea! this should really help prevent bugs like #2410 sneaking in!

**Comment by Borda** [comment]
@awaelchli any idea why Windows does not have any new PyTorch?
```
ERROR: Could not find a version that satisfies the requirement torch>=1.3 (from -r ./requirements/base.txt (line 4)) (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2)
ERROR: No matching distribution found for torch>=1.3 (from -r ./requirements/base.txt (line 4))
```
on the other hand, complete testing works fine... maybe because it is cached?

**Comment by Borda** [comment]
this seems like a Win issue with PT...
```
pytorch_lightning\core\__init__.py:339: in <module>
    from pytorch_lightning.core.decorators import data_loader
pytorch_lightning\core\decorators.py:4: in <module>
    import torch
C:\hostedtoolcache\windows\Python\3.7.7\x64\lib\site-packages\torch\__init__.py:81: in <module>
    ctypes.CDLL(dll)
C:\hostedtoolcache\windows\Python\3.7.7\x64\lib\ctypes\__init__.py:364: in __init__
    self._handle = _dlopen(self._name, mode)
E   OSError: [WinError 126] The specified module could not be found
```

**Comment by Borda** [comment]
@awaelchli @williamFalcon mind check the final version?

File: docs/source/weights_loading.rst [hunk 1]
```diff
@@ -108,9 +108,8 @@ But if you don't want to use the values saved in the checkpoint, pass in your ow
 
         def __init__(self, in_dim, out_dim):
             super().__init__()
-            self.in_dim = in_dim
-            self.out_dim = out_dim
```
Comment (awaelchli) [inline docs/source/weights_loading.rst:112]: can we not leave it here?
save_hyperparameters will only save in_dim and out_dim, right?
the user should still be able to assign whatever they want to self.

File: pytorch_lightning/loggers/neptune.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 from argparse import Namespace
 from typing import Optional, List, Dict, Any, Union, Iterable
 
-from PIL.Image import Image
+# from PIL.Image import Image
```
Comment (awaelchli) [inline pytorch_lightning/loggers/neptune.py:8]:
```suggestion
```

File: pytorch_lightning/trainer/trainer.py [hunk 3]
```diff
@@ -76,6 +76,63 @@ class Trainer(
     TrainerDeprecatedAPITillVer0_9,
     TrainerDeprecatedAPITillVer0_10,
 ):
+    """
+    Example:
+
+        >>> import torch
+        >>> from torch.nn import functional as F
+        >>> from torch.utils.data import Dataset, DataLoader
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:84]: if you want, you can move these to the top of the file, they don't necessarily need to be inside the doctest.

File: pytorch_lightning/trainer/trainer.py [hunk 4]
```diff
@@ -76,6 +76,63 @@ class Trainer(
...
+        ...     def __len__(self):
+        ...         return len(self.input_seq)
+        ...
+        ...     def __getitem__(self, item):
+        ...         return self.input_seq[item], self.output_seq[item]
+        ...
+        >>> train_loader = DataLoader(SimpleDataset(), batch_size=8)
+        >>> #define trainer and run
+        >>> model = SimpleModel()
+        >>> trainer = Trainer(max_epochs=1, progress_bar_refresh_rate=0)
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:126]:
```suggestion
        >>> model = SimpleModel()
        >>> # Define Trainer and fit model
        >>> trainer = Trainer(max_epochs=1, progress_bar_refresh_rate=0)
```

File: pytorch_lightning/trainer/trainer.py [hunk 5]
```diff
@@ -76,6 +76,63 @@ class Trainer(
...
 ):
+    """
+    Example:
+
+        >>> import torch
+        >>> from torch.nn import functional as F
+        >>> from torch.utils.data import Dataset, DataLoader
+        >>> class AAA:
+        ...     def __init__(self):
+        ...         self.a = 1
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:87]: is this needed?

**Comment by awaelchli** [comment]
it seems since you removed circle CI, we cannot preview docs anymore directly in the CI artifacts. Maybe we can add it back later.

File: pytorch_lightning/trainer/trainer.py (hunk 3 above)
Comment (Borda) [reply pytorch_lightning/trainer/trainer.py:84]: but they are not called outside test flake8 may complain about unused impprts

File: pytorch_lightning/trainer/trainer.py (hunk 5 above)
Comment (Borda) [reply pytorch_lightning/trainer/trainer.py:87]: good cacth :]

**Comment by Borda** [comment]
> it seems since you removed circle CI, we cannot preview docs anymore directly in the CI artifacts. Maybe we can add it back later.

well we are exporting build-docs with GH action but it is zip format, we may check if there is a way to browse it online or add CircleCI just for the docs? not sure how many people really used it... @PyTorchLightning/core-contributors?

File: docs/source/weights_loading.rst (hunk 1 above)
Comment (Borda) [reply docs/source/weights_loading.rst:112]: yes, he does but we do not want to recommend this practice, right? @williamFalcon

Comment (williamFalcon) [reply docs/source/weights_loading.rst:112]: people can do whichever but it’s better to use the save call for hparams tracking

**Pull request merged**